portfolio/src/main/java/com/google/sps/servlets/BookDataServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handle survey's data. Store the votes or load them. */
+@WebServlet("/book-data")
+public class BookDataServlet extends HttpServlet {
+
+  private Map<String, Integer> bookVotes = new HashMap<>();
```

Comment:
Are you sure keeping this in memory is all you need? AppEngine can easily decide to restart your server, and will then lose all the votes.
Prefer to store the votes in persistent storage instead.